### PR TITLE
[SNMP] Process enterprise OID in V1TRAP PDUs

### PR DESCRIPTION
**Pull request by sharo1k**
This request fixes omission of V1TRAP PDU event.

SNMP4J package sends SNMP V1 Traps in objects of `PDUv1` type. Class `PDUv1` is derived from `PDU` class, specific just for V1 Traps.  If using parent PDU class, there is no access to additional information - `Enterprise OID` / `GenericTrap(GT)` / `SpecificTrap(ST)`.  The code change just calls one more time `updateChannels` with that additional information. 
Regarding GenericTrap/SpecificTrap codes - if GT==6, ST must be used. However RFC1157 allows ST to be populated even if GT !=6. Possible solutions for differentiation between GT/ST are:

- if GT==6 => use ST (possible drawback - misinterpretation of traps that can have either ST or GT values)
- if ST==0 => use GT (possible drawback - SNMP implementations that doesn't set ST=0 when unused)
- use GT*Factor+ST (possible overflow, although very unlikely; has to be documented, as it may be overlooked) 

In this request I choose first option, as it seems most drawback free.
Tested with my UPS (Eaton Network-Card-MS) - traps are caught and processed further (Items updated, rules triggered).

Additional info: [OpenHAB Community discussion](https://community.openhab.org/t/solved-commit-needed-snmp-binding-v2-is-ignoring-enterprise-oid-v1trap-data/86599)

### CODE REVIEW

**Comment by sharo1k** [comment]
My bad. It seems that I pushed my temporary fix when signing.
Also replaced '6' with PDUv1.ENTERPRISE_SPECIFIC for clarity.

**Comment by J-N-K** [comment]
@sharo1k Can you follow @cweitkamp's comment and apply the code-style guidelines? Thank you.

**Comment by sharo1k** [comment]
> @sharo1k Can you follow @cweitkamp's comment and apply the code-style guidelines? Thank you.

Hi Jan, I already fixed the formatting in the previous commit - spotless check was successful. Am I missing something else?

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java [hunk 1]
```diff
@@ -208,7 +209,17 @@ public void processPdu(@Nullable CommandResponderEvent event) {
         final PDU pdu = event.getPDU();
         final String address = ((UdpAddress) event.getPeerAddress()).getInetAddress().getHostAddress();
         final String community = new String(event.getSecurityName());
-
+        
+        if ((pdu.getType() == PDU.V1TRAP) && config.community.equals(community) && (pdu instanceof PDUv1)){
```
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java]: Is the formatter really not putting a space here?
```suggestion
        if ((pdu.getType() == PDU.V1TRAP) && config.community.equals(community) && (pdu instanceof PDUv1)) {
```

Comment (sharo1k) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java]: Strange, but spotless finds problems only in xml files. It should be fine now

**Comment by J-N-K** [comment]
Travis fails unrelated and previous builds succeeded.

**Pull request merged**